x-pack/plugins/fleet/server/routes/agent_policy/handlers.ts

```diff
@@ -461,3 +462,31 @@ export const downloadK8sManifest: FleetRequestHandler<
     return defaultFleetErrorHandler({ error, response });
   }
 };
+
+export const getUninstallTokensByPolicyIdHandler: FleetRequestHandler<
+  TypeOf<typeof GetUninstallTokensByPolicyIdRequestSchema.params>
+> = async (context, request, response) => {
+  const uninstallTokenService = appContextService.getUninstallTokenService();
+  if (!uninstallTokenService) {
+    return response.customError({
+      statusCode: 500,
+      body: { message: 'Uninstall Token Service is unavailable.' },
+    });
+  }
+
+  try {
+    const { agentPolicyId } = request.params;
+
+    const tokensForOnePolicy = await uninstallTokenService.getTokenHistoryForPolicy(agentPolicyId);
+
+    if (tokensForOnePolicy.total === 0) {
+      return response.notFound({
```

Comment:
it probably make sense to return an empty list here?

Reply:
every policy should have (at least) one uninstall token. if there is none, it means that either the policyID is wrong (but we cannot validate this, because we hold tokens for deleted policies, too, that don't exist elsewhere anymore), or somehow there is no token generated for it. either way, I thought this is a special case that could be emphasised with 404.

the other route, `GET /api/fleet/uninstall_tokens?policyId=xxx`, which can be used for filtering tokens even on partial `policyId`, returns an empty list if no tokens are found, because it's more like a search than reading a resource.

that's the thought behind the 404, what do you think of this?

Reply:
If I'm not wrong, we add the uninstall tokens to all policies (existing and new ones) even those don't have the feature enabled. @joeypoon can confirm that.

Reply:
currently we either generate a token for all policies or no policies depending on the feature flag `agentTamperProtectionEnabled`.